Display the events in a list view.

pandilladivya opened the pull request:
The following is the demo of the work done.

![ezgif com-video-to-gif 3](https://user-images.githubusercontent.com/16818129/27218128-7667c86a-529a-11e7-80e3-d902c9c6019f.gif)


Used data from:
https://github.com/systers/mobile-app/issues/4 and
https://github.com/systers/mobile-app/issues/5

pandilladivya commented:
@HimanshiJain @sandarumk @MeepyMay 
Kindly Review.
Thanks 😊

HimanshiJain commented:
The appearance of each event item does not look that good to me. Try different font sizes and maybe less spacing between the details.

HimanshiJain commented:
And also how would an event that does require volunteer look like? Manage the appearance and space considering that too.

sandarumk commented:
@pandilladivya Are you done with the changes suggested by @HimanshiJain ? Once the previous PR is merged, fix the merge conflicts on this PR too

pandilladivya commented:
@HimanshiJain We shall use card views for showing schedules and List views for showing events.
However I've changed the UI a little after reading your suggestions. 😊 


<img width="366" alt="screen shot 2017-07-20 at 11 13 50 pm" src="https://user-images.githubusercontent.com/16818129/28431651-ae9aacfc-6da2-11e7-94cf-f3d1b0b04914.png">


If you have anymore suggestions please let me know.
@sandarumk I've made the changes suggested by Himanshi and also resolved conflicts 😄 
Kindly review this PR.
Thank you

HimanshiJain commented:
@pandilladivya Better than before. Although still not too much convinced. Wouldn't it have a lot of empty space if "Required Volunteers" is not applicable for an event? Also, we can have a list of cards. Did we fix something like card views for showing schedules and List views for showing events? @sandarumk what do you think?

pandilladivya commented:
Okay. If you want me to put cardview. I'm fine with it 🙂

pandilladivya commented on app/src/main/AndroidManifest.xml:
```diff
@@ -0,0 +1,73 @@
+
+    <uses-permission android:name="android.permission.INTERNET"/>
+    <application
+        android:allowBackup="true"
```
Yes.

HimanshiJain replied:
So do we need to set this explicitly?

pandilladivya replied:
This came by default when I created the basic android project. And I have seen many projects for which this was set. So I didn't remove it.

pandilladivya commented on app/src/main/AndroidManifest.xml:
```diff
@@ -0,0 +1,73 @@
+    <uses-permission android:name="android.permission.INTERNET"/>
+    <application
+        android:allowBackup="true"
+        android:icon="@mipmap/ic_launcher"
```
I think @MeepyMay can give more insight about this.

pandilladivya commented on app/src/main/java/vola/systers/com/volunteers_android/fragments/EventsMapFragment.java:
```diff
@@ -0,0 +1,44 @@
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+
```
Sure.

pandilladivya commented on app/src/main/java/vola/systers/com/volunteers_android/fragments/EventsListFragment.java:
```diff
@@ -0,0 +1,166 @@
+    static String startDate, endDate, id,name,startTime,endTime;
+
+    // URL to get events JSON
+    private static String url = "http://divya-gsoc.esy.es/sample/data.json";
```
This URL is just for temporary purpose, this was before May give me etouches Credentials. So, In coming PRs we will have to change everything and since I've hosted in somewhere, I don't think it is very important to rechange the url and host it. 

And I agree with changing the name of it from url to something meaningful :)

HimanshiJain replied:
So should not we check in the updated URLs that will be finally used?

pandilladivya replied:
This was just the mock data. 
Later in other PRs I've integrated etouches API, So removed these URLs and used HttpPost and Get requests.
Now, since then we decided not to use etouches, We need to get these event data from firebase database. So, I think there will not be any such updated URL maybe.
Please do correct me if I misunderstood your question.

HimanshiJain replied:
Right. So should we not change the code to get this data from Firebase and then check in this code?

pandilladivya replied:
Yes. This was the code written in Phase - I. in mid june. Back then my task was to integrate the mock json data which was given by May in 
https://github.com/systers/mobile-app/issues/4 and
https://github.com/systers/mobile-app/issues/5 .

There was no firebase integrated in the app at this point of time.
In the Next PRs firebase was integrated.
So we cant do the firebase integration in this PR.

pandilladivya replied:
This is the link to the updated version of this PR with etouches Integration. But had to close since we are not using it now.

https://github.com/systers/volunteers-android/pull/37

Once the firebase db is created by May, I can update the code to get these data from the db and create a PR then for it :)

pandilladivya commented on app/src/main/java/vola/systers/com/volunteers_android/activities/Menu.java:
```diff
@@ -0,0 +1,320 @@
+
+    // flag to load home fragment when user presses back key
+    private boolean shouldLoadHomeFragOnBackPress = true;
+    private Handler handler;
```
In homePage, the fragment is loaded with cross fade effect using runnable.And if pending_runnable is not null, then it adds to the message queue.

And Yes, I'll put more meaningful name :)

pandilladivya commented on app/src/main/AndroidManifest.xml:
```diff
@@ -0,0 +1,73 @@
+        android:icon="@mipmap/ic_launcher"
+        android:label="@string/app_name"
+        android:roundIcon="@mipmap/ic_launcher_round"
+        android:supportsRtl="true"
```
Yes, It declares whether our application is willing to support right-to-left (RTL) layouts. Since our targetSdkVersion is greater than 17 , various RTL APIs will be activated and used by the system so your app can display RTL layouts.

sandarumk commented on app/src/main/java/vola/systers/com/volunteers_android/fragments/EventsListFragment.java:
```diff
@@ -0,0 +1,166 @@
+                        endTime=eventDetailsJsonObject.getString("endtime");
+
+                        // tmp hash map for single event
+                        HashMap<String, String> event = new HashMap<>();
```
Agreed. Good idea.

pandilladivya replied:
Noted.

HimanshiJain commented on app/src/main/java/vola/systers/com/volunteers_android/adapter/NavDrawerListAdapter.java:
```diff
@@ -0,0 +1,69 @@
+package vola.systers.com.volunteers_android.adapter;
```
Agreed. It should be the domain name in reverse order followed by name.

pandilladivya replied:
Can we change this later ?

sandarumk replied:
I think this is a fundamental change which need to be done and merged ASAP. If not for the given PRs can you make a new PR just to change this in the existing code? Or else do this in the improvements in this PR.

pandilladivya replied:
Actually, in https://github.com/systers/volunteers-android/pull/38 i had to change the package name to match the one given in systers's firebase account.

pandilladivya replied:
If we decide to change the package name, we need to update it in the firebase account too. So I think we would require @MeepyMay 's help to do that.

sandarumk replied:
So does the package name vola.systers.com.android is in the systers firebase account? @MeepyMay Can we change it. Usually the package name is the reverse domain nam. https://stackoverflow.com/questions/6273892/android-package-name-convention

pandilladivya replied:
In Firebase's account, it is **vola.systers.com.android**

HimanshiJain commented on app/src/main/java/vola/systers/com/volunteers_android/handler/HttpHandler.java:
```diff
@@ -0,0 +1,63 @@
+import java.net.ProtocolException;
+import java.net.URL;
+
+public class HttpHandler {
```
Right, because there is no state associated with each object of the class.

pandilladivya replied:
Yes. Agreed to that. I'll modify

pandilladivya commented on app/src/main/java/vola/systers/com/volunteers_android/activities/Menu.java:
```diff
@@ -0,0 +1,320 @@
+import vola.systers.com.volunteers_android.fragments.NavigateFragment;
+
+
+public class Menu extends AppCompatActivity {
```
Agreed. Will Rename!

pandilladivya commented on app/src/androidTest/java/vola/systers/com/volunteers_android/ExampleInstrumentedTest.java:
```diff
@@ -0,0 +1,26 @@
+package vola.systers.com.volunteers_android;
+
```
No. It was given by default. I'll delete it

pandilladivya commented on app/src/main/res/layout/list_item.xml:
```diff
@@ -0,0 +1,163 @@
+    android:paddingLeft="@dimen/activity_horizontal_margin"
+    >
+
+    <RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Oh! Yes. I think one relative view is redundant.. 
I'll remove it. Thanks for pointing out :)

pandilladivya commented on app/src/main/res/layout/list_item.xml:
```diff
@@ -0,0 +1,163 @@
+        android:focusable="True"
+        >
+
+        <LinearLayout
```
Yeah. It works the same without linear layout too. Will remove that layout ☺️

pandilladivya commented:
@sandarumk @HimanshiJain 

I've implemented all the suggested changes.

1. Added Utils Package and made the HttpHandler a util class
2. Renamed Menu to MenuActivity
3.  Removed HashMap and created a separate event class.
4. Created a separate Custom Handler
5. Changed names of variables to specify their purpose.
6. Made Layout changes.
7. Deleted Example Test.

Kindly review

sandarumk commented:
@pandilladivya  In an overall sense, commit messages should be meaningful. For example commits like events list should change. https://chris.beams.io/posts/git-commit/. If you can, please try not to include the merge commits. Do a rebase if necessary rather than merge. In that way it will be more neat.
@HimanshiJain once you are done with the review and approved changes, please merge do a squash and merge. So that it will remove the merge commits.
